assets/js/common.js

```diff
@@ -554,6 +562,41 @@ var hivepress = {
 								fields.eq(0).val(formattedDates[0]);
 								fields.eq(1).val(formattedDates[1]);
 							} else {
+								var days = instance.calendarContainer.querySelectorAll('.flatpickr-day'),
+									previousLastDayIndex = null,
+									nextLastDayIndex = null;
+
+								days.forEach((day, index) => {
+									if (day.dateObj.getTime() === selectedDates[0].getTime() && day.classList.contains('last-day')) {
+										instance.clear();
+										return;
+									}
+
+									if (day.dateObj < selectedDates[0]) {
+										if (day.classList.contains('last-day')) {
+											previousLastDayIndex = index;
+										}
+									} else if (day.dateObj > selectedDates[0]) {
+										if (day.classList.contains('last-day') && nextLastDayIndex === null) {
```

Comment:
If I understand correctly by comparing it with null you check that it's the first occurance, but instead you can just exit the loop when assigning the nextLastDayIndex below, we don't need to check further dates after we find it.

Overall, after implementing the fixes described above, please make sure that the code is as optimized as possible while being easy to understand